src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Factory/EntityTypeFactory.php

```diff
@@ -119,23 +119,34 @@ public function createEditForm(ManageEntity $entity, Service $service, string $e
                 return $this->formFactory->create(
                     OidcngResourceServerEntityType::class,
                     $command,
-                    $this->buildOptions($environment)
+                    $this->editBuildOptions($environment)
                 );
             case (Constants::TYPE_OAUTH_CLIENT_CREDENTIAL_CLIENT):
                 $command = $this->saveCommandFactory->buildOauthCccCommandByManageEntity($entity, $environment, $isCopy);
                 $command->setService($service);
-                return $this->formFactory->create(OauthClientCredentialEntityType::class, $command, $this->buildOptions($environment));
+                return $this->formFactory->create(OauthClientCredentialEntityType::class, $command, $this->editBuildOptions($environment));
         }
 
         throw new InvalidArgumentException("invalid form type requested");
     }
 
-    private function buildOptions($environment)
+    private function createBuildOptions($environment)
     {
         $options = [];
         if ($environment === Constants::ENVIRONMENT_PRODUCTION) {
             $options = ['validation_groups' => ['Default', 'production']];
         }
         return $options;
     }
+
+    private function editBuildOptions($environment)
+    {
+        $options = [];
+        if ($environment === Constants::ENVIRONMENT_PRODUCTION) {
+            $options = ['validation_groups' => ['Default', 'production'],
+                'publish_button_label' => 'Change request',
```

Comment:
Shouldn't this text be translatable? I see this used to be a hard-coded text before your change. But still..

Reply:
Agree, but perhaps better to check all buttons and make a general change, for example in the upgrade SF 5